File: initializer.py

#initializer:constructor

#Speical method in a class
#initializer: method is called whenver the blueprint is used
#To create an object
#method function is inside a class

#method your create you have to hhave thhe self: keyworkd

#how to create a class <-->What a class is
# initialiZer <constrocture>
# self key what it is :<this>

class Human():

    def __init__(self,gender,name):
        print("The initializer wass called")
        self.gender=gender
        self.name=name
        if self.gender=="Male":
            self.ribs=24
            self.curse="Suffer"
        else :
          self.ribs=23
          self.curse="Pain"
        

    def print_self(self):
        print("----------------------")
        print("name",self.name)
        print("gender",self.gender)
        print("ribs",self.ribs)
        print("curse",self.curse)
        print("---------------------")


# adam=Human(name="adam",gender="Male") #object from a class
adam=Human(name="adam",gender="Male")
adam.print_self()

eve=Human(name="eve",gender="Female")
eve.print_self()
# eve=Human(name="eve",gender="Female")
# print("name",eve.name)
# print("gender",eve.gender)
# print("ribs",eve.ribs)
# print("curse",eve.curse)

File: initializer.js
class Human {
  constructor(gender, name) {
    this.gender = gender;
    this.name = name;

    if (gender === "Male") {
      this.ribs = 24;
      this.curse = "Suffer";
    } else {
      this.ribs = 23;
      this.curse = "Pain";
    }
  }

  print_self() {
    console.log("----------------------");
    console.log("name", this.name);
    console.log("gender", this.gender);
    console.log("ribs", this.ribs);
    console.log("curse", this.curse);
    console.log("---------------------");
  }
}

let adam = new Human("Male", "Adam");
adam.print_self();

let eve = new Human("Female", "Eve");
eve.print_self();